src/index.ts

```diff
@@ -344,16 +372,18 @@ export function graphqlHTTP(options: Options): Middleware {
         });
       }
 
-      // Collect and apply any metadata extensions if a function was provided.
-      // https://graphql.github.io/graphql-spec/#sec-Response-Format
+      if (isAsyncIterable(executeResult)) {
+        // Get first payload from AsyncIterator. http status will reflect status
+        // of this payload.
+        const asyncIterator = getAsyncIterator<ExecutionResult>(executeResult);
+        const { value } = await asyncIterator.next();
```

Comment:
`await` can throw and should be inside `try` 🔼

Reply:
test for that would be great. You can do that by using custom `executeFn` that return an async iterator.

Reply:
@IvanGoncharov do you think it makes sense for the `try/catch` that is currently wrapping `await executeFn({...})` to be expanded to also wrap this?

Reply:
@robrichard Yes, I think it would work 👍